### DISCUSSION TITLE
Welcome to otel-java-basics Discussions!

### DISCUSSION THREAD

**Discussion by tedsuo** [tedsuo/otel-java-basics#3] [General]
## 👋 Welcome!
Have questions about OpenTelemetry, or about how this example works? Add a new discussion and I'll be happy to help.

**Comment by sourac** [comment]
Hi, @tedsuo I'm working on a microservice architecture where at the start, we're receiving the request with an additional header field as servicePlanNumber, I want to use baggage propagation to pass this servicePlanNumber to all the downstream services explicitly.

Right now, I'm using  Opentelemetry java agent, which does the creation of trace-id/span-id automatically, all I want is to have the servicePlanNumber passed downstream and added to the logs along with the trace-id/span-id. earlier using spring sleuth baggage was pretty straightforward but I need help with this. I'm using Spring boot version 3.0.0.

Sender Service:

`    @GetMapping("/api")
    public Flux<String> span(@RequestHeader(value = "servicePlanNumber", required = false) String servicePlanNumber) {
        log.info("Inside span of MicroOne");

        Tracer tracer = GlobalOpenTelemetry.get().getTracer("/api");
        Baggage baggage = Baggage.builder().put("servicePlanNumber", servicePlanNumber).build();
        Context context=Context.current().with(baggage);
        Span span = tracer.spanBuilder("/api").setParent(context).startSpan();

        try( Scope scope = span.makeCurrent()) {
            Baggage baggage1 = Baggage.fromContext(context);
            log.info("baggage value : " + baggage1.getEntryValue("servicePlanNumber"));
            log.info("About to hit...");
            log.info("context {} ", context);
            return builder.baseUrl("http://localhost:9433")
                    .build()
                    .get()
                    .uri("/MicroTwo/api")
                    .retrieve()
                    .bodyToFlux(String.class);

        }finally {
            span.end();
        }
    }`


Receiver Service:

`    @GetMapping("/api")
    public Flux<String> span() {
        log.info("Inside MicroTwo span");

      Tracer tracer = GlobalOpenTelemetry.get().getTracer("/api");
      log.info("tracer : {}", tracer );
      Context context = Context.current();
        log.info("context : {}",context);
        Baggage baggage = Baggage.fromContext(context);
        String servicePlanNumber = baggage.getEntryValue("servicePlanNumber");
        log.info("servicePlanNumber : {} ", servicePlanNumber);
        return Flux.fromIterable(List.of("One","Inland","Ocean","Truck"))
                .delayElements(Duration.ofSeconds(0));

    }`. But i'm not getting the value servicePlanNumber in this service, also how do i add the this servicePlanNumber in all the logs generated by otel java agent.
